packages/core/useDeviceOrientation/index.ts

```diff
@@ -1,20 +1,24 @@
 /* this implementation is original ported from https://github.com/logaretm/vue-use-web by Abdelrahman Awad */
 
+import { isClient } from '@vueuse/shared'
 import { ref, Ref } from 'vue-demi'
 import { useEventListener } from '../useEventListener'
 
-export function useDeviceOrientation() {
+export function useDeviceOrientation(target: Window | null = isClient ? window : null) {
+  const eventTarget = isClient ? target : null
```

Comment:
Let's just use target

Reply:
I was analyzing here, we have many places that would need this implementation. Maybe not the best way. And I will think about it better.

Reply:
For example: `useDeviceMotion`, `useDeviceLIght`

Reply:
Yeah, it's better to have an option for it. Will also be more extensible for us to add more options in the future.

I think we can make a shared interface with shared js-docs (providing more details/explication for users) like

```ts
export interface ConfigurableWindow {
  /*
   * Specify a custom `window` instance, e.g. working with iframes or in testing envroiments.
   * ...(more)
   */
  window?: Window
}

export interface ConfigurableDocument {
  /*
   * Specify a custom `document` instance, e.g. working with iframes or in testing envroiments.
   * ...(more)
   */
  document?: Document
}
```

```ts
interface DeviceMotionOptions extends ConfigurableWindow {
  throttleMs: number
}

export function useDeviceMotion(options: DeviceMotionOptions = { throttleMs: 10 }) {
  // options.window
  /* ... */
}
```

Reply:
`navigator`, `location` as well. Any global reference in functions should be configurable then I think.

Reply:
I agree with you, I was looking for a solution as a global configuration, but I think your idea will give more flexibility for the use of the functions.

Reply:
I will do the refactor 👍